src/decoder/mod.rs

```diff
@@ -895,6 +898,18 @@ impl<R: Read + Seek> Decoder<R> {
         self.get_tag(tag)?.into_string()
     }
 
+    pub fn tag_iter(&mut self) -> impl Iterator<Item = TiffResult<(Tag, ifd::Value)>> + '_ {
```

Comment:
Could you also add a short doc comment explaining what this method does? Something like "Returns an iterator over all tags in the current image"

Reply:
Sure!